src/main/webapp/js/studentProfile.es6

```diff
@@ -1,47 +1,101 @@
+    const scaledHeight = picture.prop('naturalHeight') * transformData.scale;
+
+    $('#cropBoxLeftX').val(transformData.x);
+    $('#cropBoxTopY').val(transformData.y);
+    $('#cropBoxRightX').val(transformData.x + transformData.w);
+    $('#cropBoxBottomY').val(transformData.y + transformData.h);
+    $('#rotate').val(transformData.angle);
+    $('#pictureWidth').val(scaledWidth);
+    $('#pictureHeight').val(scaledHeight);
+    $('#profilePictureEditForm').submit();
+}
+
+function finaliseUploadPictureForm() {
+    if ($('#studentPhoto').val() === '') {
+        return;
+    }
+
+    const initialSubmitMessage = $('#profileUploadPictureSubmit').html();
+    $.ajax({
+        url: `/page/studentProfileCreateFormUrl?user=${$("input[name='user']").val()}`,
+        beforeSend() {
+            $('#profileUploadPictureSubmit').html('<img src="/images/ajax-loader.gif">');
+        },
+        error() {
+            $('#profileUploadPictureSubmit').text(initialSubmitMessage);
+            setStatusMessage('There seems to be a network error, please try again later', StatusType.DANGER);
+            scrollToTop({ duration: '' });
+        },
+        success(data) {
+            if (data.isError) {
+                $('#profileUploadPictureSubmit').text(initialSubmitMessage);
+                setStatusMessage('There seems to be a network error, please try again later', StatusType.DANGER);
+                scrollToTop({ duration: '' });
+            } else {
+                $('#profilePictureUploadForm').attr('enctype', 'multipart/form-data');
+                // for IE compatibility
+                $('#profilePictureUploadForm').attr('encoding', 'multipart/form-data');
+                $('#profilePictureUploadForm').attr('action', data.formUrl);
+                $('#profilePictureUploadForm').submit();
+            }
+        },
+    });
+}
+
+$(document).ready(() => {
+    $('.form-control').on('click', function () {
         if ($(this).val() === $(this).attr('data-actual-value')) {
             $(this).select();
         }
     });
 
-    $('#profileUploadPictureSubmit').on('click', function() {
+    $('#profileUploadPictureSubmit').on('click', () => {
         finaliseUploadPictureForm();
     });
 
-    $('#profileEditPictureSubmit').on('click', function() {
+    $('#profileEditPictureSubmit').on('click', () => {
         finaliseEditPictureForm();
     });
 
-    $(window).load(function() {
-        $('#studentPhoto').change(function() {
+    $(window).load(() => {
+        $('#studentPhoto').change(function () {
             if ($(this).val() === '') {
                 $('#profileUploadPictureSubmit').prop('disabled', true);
                 $('.filename-preview').val('No File Selected');
             } else {
                 $('#profileUploadPictureSubmit').prop('disabled', false);
-                $('.filename-preview').val($(this).val().split('\\').pop().split('/').pop());
+                const val = $(this)
+                    .val()
```

Comment:
@li-kai are you fixing this? 😛
We can go perfectly fine with what we have right now, so no pressure.